zap/zap.go

```diff
@@ -14,89 +15,176 @@ var (
+		return false
+	}
+
+	return zpl.logger.Check(zpl.logger.Level(), "") != nil
 }
 
 // WithEnabled passes the logger and if it's enabled
-func (zl *Logger) WithEnabled() (slog.Logger, bool) {
-	return zl, true
+func (zpl *Logger) WithEnabled() (slog.Logger, bool) {
+	return zpl, zpl.Enabled()
 }
 
 // Print adds a log entry with arguments handled in the manner of fmt.Print
-func (zl *Logger) Print(args ...any) {
-	zl.print(fmt.Sprint(args...))
+func (zpl *Logger) Print(args ...any) {
+	if zpl.Enabled() {
+		zpl.print(fmt.Sprint(args...))
+	}
 }
 
 // Println adds a log entry with arguments handled in the manner of fmt.Println
-func (zl *Logger) Println(args ...any) {
-	zl.print(fmt.Sprintln(args...))
+func (zpl *Logger) Println(args ...any) {
+	if zpl.Enabled() {
+		zpl.print(fmt.Sprintln(args...))
+	}
 }
 
 // Printf adds a log entry with arguments handled in the manner of fmt.Printf
-func (zl *Logger) Printf(format string, args ...any) {
-	zl.print(fmt.Sprintf(format, args...))
+func (zpl *Logger) Printf(format string, args ...any) {
+	if zpl.Enabled() {
+		zpl.print(fmt.Sprintf(format, args...))
+	}
 }
 
-func (zl *Logger) print(string) {}
+func (zpl *Logger) print(msg string) {
+	msg = strings.TrimSpace(msg)
+	if ce := zpl.logger.Check(zpl.logger.Level(), msg); ce != nil {
+		ce.Write()
+	}
+
+}
 
 // Debug returns a new logger set to add entries as level Debug
-func (zl *Logger) Debug() slog.Logger {
-	return zl.WithLevel(slog.Debug)
+func (zpl *Logger) Debug() slog.Logger {
+	return zpl.WithLevel(slog.Debug)
 }
 
 // Info returns a new logger set to add entries as level Info
-func (zl *Logger) Info() slog.Logger {
-	return zl.WithLevel(slog.Info)
+func (zpl *Logger) Info() slog.Logger {
+	return zpl.WithLevel(slog.Info)
 }
 
 // Warn returns a new logger set to add entries as level Warn
-func (zl *Logger) Warn() slog.Logger {
-	return zl.WithLevel(slog.Warn)
+func (zpl *Logger) Warn() slog.Logger {
+	return zpl.WithLevel(slog.Warn)
 }
 
 // Error returns a new logger set to add entries as level Error
-func (zl *Logger) Error() slog.Logger {
-	return zl.WithLevel(slog.Error)
+func (zpl *Logger) Error() slog.Logger {
+	return zpl.WithLevel(slog.Error)
 }
 
 // Fatal returns a new logger set to add entries as level Fatal
-func (zl *Logger) Fatal() slog.Logger {
-	return zl.WithLevel(slog.Fatal)
+func (zpl *Logger) Fatal() slog.Logger {
+	return zpl.WithLevel(slog.Fatal)
 }
 
 // WithLevel returns a new logger set to add entries to the specified level
-func (zl *Logger) WithLevel(level slog.LogLevel) slog.Logger {
-	return zl
+func (zpl *Logger) WithLevel(level slog.LogLevel) slog.Logger {
+
+	var levels = []zapcore.Level{
+		slog.UndefinedLevel: zapcore.InvalidLevel,
+		slog.Fatal:          zapcore.FatalLevel,
+		slog.Error:          zapcore.ErrorLevel,
+		slog.Warn:           zapcore.WarnLevel,
+		slog.Info:           zapcore.InfoLevel,
+		slog.Debug:          zapcore.DebugLevel,
+	}
+
+	if level < -1 || int(level) >= len(levels) {
```

Comment:
here `level` is a `slog.Level` so `-1` isn't valid.

Reply:
fixed